pkg/autoscaler/http_scrape_client_test.go

```diff
@@ -0,0 +1,189 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package autoscaler
+
+import (
+	"bytes"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"testing"
+	"time"
+)
+
+const (
+	testURL = "http://test-revision-metrics.test-namespace:9090/metrics"
+
+	testAverageConcurrencyContext = `# HELP queue_average_concurrent_requests Number of requests currently being handled by this pod
```

Comment:
Nothing like good ol' text format.
Does prometheus support binary by any chance?

Reply:
I think so but didn't look into what/how to use. Added a TODO anyway.

Reply:
Nope, there's only a text protocol. See https://github.com/prometheus/docs/blob/master/content/docs/instrumenting/exposition_formats.md